_posts/2025-01-07-arrow-result-transfer.md

```diff
@@ -0,0 +1,124 @@
+
+Yet many query results today continue to flow through legacy APIs and protocols that add massive serialization and deserialization (“ser/de”) overheads by forcing data into inefficient transfer formats. In a [2021 paper](https://www.vldb.org/pvldb/vol14/p534-li.pdf), Tianyu Li et al. presented an example using ODBC and the PostgreSQL protocol in which 99.996% of total query time was spent on ser/de. That is arguably an extreme case, but we have observed 90% or higher in many real-world cases. Today, for data engineering and data analytics queries, there is a strong incentive to choose a transfer format that speeds up ser/de.
+
+Enter Arrow.
+
+The Apache Arrow open source project defines a [data format](https://arrow.apache.org/docs/format/Columnar.html) that is designed to speed up—and in some cases eliminate—ser/de in query result transfer. Since its creation in 2016, the Arrow format and the multi-language toolbox built around it have gained widespread use, but the technical details of how Arrow is able to slash ser/de overheads remain poorly understood. To help address this, we outline five key attributes of the Arrow format that make this possible.
+
+### 1. The Arrow format is columnar.
+
+In a columnar (column-oriented) data format, the values for each column in the data are held in contiguous blocks of memory. This is in contrast to row-oriented formats, in which the values for each row are held in contiguous blocks of memory.
+
+<figure style="text-align: center;">
+  <img src="{{ site.baseurl }}/img/arrow-result-transfer/part-1-figure-1-row-vs-column-layout.png" width="100%" class="img-responsive" alt="Figure 1: An illustration of row-oriented and column-oriented physical memory layouts of a logical table containing three rows and five columns.">
+  <figcaption>Figure 1: An illustration of row-oriented and column-oriented physical memory layouts of a logical table containing three rows and five columns.</figcaption>
+</figure>
+
+High-performance analytic databases, data warehouses, query engines, and storage systems have converged on columnar architecture because it speeds up the most common types of analytic queries. Examples of modern columnar query systems include Amazon Redshift, ClickHouse, Databricks Photon Engine, DuckDB, Google BigQuery, Microsoft Azure Synapse Analytics, Snowflake, and Voltron Data Theseus.
```

Comment:
Add Apache DataFusion and Vertica?

Reply:
Added DataFusion!

I left off Vertica because I'm confused about its current status. Looks like it's now been rebranded as "OpenText Analytics Database"? Happy to add it if you think it's warranted and it qualifies as "modern."

Reply:
That's interesting. It is the "OG" modern columnar analytic database that kicked off a wave of other columnar analytics DBMSs, I think it's fine to leave things as they are

Reply:
I went ahead and added it. Thanks!